src/Bindings/LuaState.h

```diff
@@ -112,6 +115,63 @@ class cLuaState
 	} ;
 	
 	
+	/** Represents a callback to Lua that C++ code can call.
+	Is thread-safe and unload-safe.
+	When the Lua state is unloaded, the callback returns an error instead of calling into non-existent code.
+	To receive the callback instance from the Lua side, use RefStack() or (better) cLuaState::GetStackValue(). */
+	class cCallback
+	{
+	public:
+		/** Creates an unbound callback instance. */
+		cCallback(void) = default;
+
+		~cCallback()
+		{
+			Clear();
+		}
+
+		/** Calls the Lua callback, if still available.
+		Returns true if callback has been called.
+		Returns false if the Lua state isn't valid anymore. */
+		template <typename... Args>
+		bool Call(Args &&... args)
```

Comment:
Ah, sorry, I forgot to push the code to that repo, I was on a train and the net wasn't working.

I wasn't sure if we were storing the lambdas as values, as references or as pointers.

Anyway, as for this PR, I don't think it's worth it, because we're not using the class as an interface of any kind, it's a regular value class and I don't think it's gonna be used directly as a callback. I could change it if you really want me to, I've tested it yesterday and it didn't seem to produce anything awkward and it's just some 4 lines of changes.